communicator/ssh/communicator.go

```diff
@@ -180,10 +181,11 @@ func (c *Communicator) Connect(o terraform.UIOutput) (err error) {
 		return err
 	}
 
-	log.Printf("[DEBUG] handshaking with SSH")
-	host := fmt.Sprintf("%s:%d", c.connInfo.Host, c.connInfo.Port)
-	sshConn, sshChan, req, err := ssh.NewClientConn(c.conn, host, c.config.config)
+	log.Printf("[DEBUG] Connection established. Handshaking for user %v", c.connInfo.User)
+	sshConn, sshChan, req, err := ssh.NewClientConn(c.conn, hostAndPort, c.config.config)
 	if err != nil {
+		err = errors.New(fmt.Sprintf("ssh %s@%s: %s", c.connInfo.User, hostAndPort, err))
```

Comment:
Thank you for improving all the debug messages - these look 👍 to me.

How do you feel about using https://github.com/hashicorp/errwrap for wrapping the error here, so we're not just turning the error into a string, but retaining all useful details?

e.g.

```go
errwrap.Wrapf(fmt.Sprintf("Failed to establish SSH connection (%s@%s): {{err}}", c.connInfo.User, hostAndPort), err)
```

Reply:
I was wondering if there was a better way.  I'll make that change.

Reply:
Done